core/contracts/financial_templates/FeePayer.sol

```diff
@@ -0,0 +1,79 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+import "../FixedPoint.sol";
+import "../Finder.sol";
+import "../Testable.sol";
+import "../StoreInterface.sol";
+
+contract FeePayer is Testable {
+    using SafeMath for uint;
+    using FixedPoint for FixedPoint.Unsigned;
+
+    /**
+     * The collateral currency used to back the positions in this contract.
+     */
+
+    IERC20 public collateralCurrency;
+
+    /**
+     * Finder contract used to look up addresses for UMA system contracts.
+     */
+    Finder public finder;
+
+    /**
+     * Tracks the last block time when the fees were paid.
+     */
+    uint public lastPaymentTime;
+
+    /**
+     * @notice modifier that calls payFees().
+     */
+    modifier fees {
+        payFees();
+        _;
+    }
+
+    constructor(address collateralAddress, address finderAddress, bool _isTest) public Testable(_isTest) {
+        collateralCurrency = IERC20(collateralAddress);
+        finder = Finder(finderAddress);
+        lastPaymentTime = getCurrentTime();
+    }
+
+    /**
+     * @notice Pays UMA DVM fees to the Store contract.
+     * @return the amount of collateral that was paid to the Store.
+     */
+
+    function payFees() public returns (FixedPoint.Unsigned memory totalPaid) {
+        StoreInterface store = StoreInterface(finder.getImplementationAddress("Store"));
+        uint time = getCurrentTime();
+        (FixedPoint.Unsigned memory regularFee, FixedPoint.Unsigned memory latePenalty) = store.computeRegularFee(
+            lastPaymentTime,
+            time,
+            pfc()
+        );
+        lastPaymentTime = time;
+
+        if (regularFee.isGreaterThan(0)) {
+            SafeERC20.safeApprove(collateralCurrency, address(store), regularFee.rawValue);
+            store.payOracleFeesErc20(address(collateralCurrency));
```

Comment:
Do we need a separate method call to `payOracleFees` if the collateral currency is ETH?

I.E. can ETH be cast to an `IERC20`?

Reply:
I think we've sorta agreed on just using wETH instead of ETH so we don't have to bother with separate ETH methods.

Reply:
that works

Reply:
Regardless, if we want to use native ETH, we can worry about adding that compatibility throughout the contract in a future PR.